zerver/views/__init__.py

```diff
@@ -1291,3 +1292,23 @@ def email_unsubscribe(request, type, token):
 
     return render_to_response('zerver/unsubscribe_link_error.html', {},
                               context_instance=RequestContext(request))
+
+# Custom realm emoji upload and deletion
+
+def list_emoji(request, user_profile):
+    return json_success({'emoji': user_profile.realm.get_emoji()})
+
+def upload_emoji(request, user_profile):
+    emoji_name = request.POST.get('name', None)
+    emoji_url = request.POST.get('url', None)
+    try:
+        do_add_realm_emoji(user_profile.realm, emoji_name, emoji_url)
+    except ValidationError as e:
+        if '__all__' in e.message_dict:
+            return json_error(e.message_dict['__all__'])
+        return json_error('Enter a valid name and URL')
+    return json_success()
+
+def delete_emoji(request, user_profile, emoji_name):
+    do_remove_realm_emoji(user_profile.realm, emoji_name)
+    return json_success({})
```

Comment:
I think probably it makes sense to put these new routes in a new zerver/views/realm_emoji.py file, given that it's we're splitting up the view files.

Also, fix the missing newline at the end of the file.